mlflow/types/utils.py

```diff
@@ -79,7 +79,90 @@ def clean_tensor_type(dtype: np.dtype):
     return dtype
 
 
-def _infer_datatype(data) -> DataType:
+def _infer_colspec_type(data: Any) -> Union[DataType, Array, Object]:
+    """
+    Infer an MLflow Colspec type from the dataset.
+
+    :param data: data to infer from.
```

Comment:
Let's use google style comment per the CONTRIBUTING.md

Reply:
Thank you for the catch!

Apparently we actually use reStructuredText style more.
- https://github.com/search?q=repo%3Amlflow%2Fmlflow+%2F%28%3F-i%29Args%3A%2F&type=code
- https://github.com/search?q=repo%3Amlflow%2Fmlflow%20%3Aparam&type=code

Wondering if we should change CONTRIBUTING.md as such, WDYT @harupy @BenWilson2?